### PR TITLE
CCPVersioner

### PR DESCRIPTION
**Pull request by macmule**
Added CCPVersioner Processor to generate installs array from downloaded media for HD & APRO installers.

TODO: Munki recipes seem fine, but JSS recipes are not complete & have some issues.

The needed .xml have been added, but there is some ways to go to pass correct information to JSS.

Issues such as:
1) Variable for name
2) Keep app bundle as-is when creating smart group criteria
3) Append version number to PKG for jss.recipes

The CFBundleShortVersionString is being imported into munki & jss correctly.

Also, Munki installs attar is using the “new” bundle path format.

### CODE REVIEW

**Comment by timsutton** [comment]
Looks good to me so far, quickly tested with PPRO!

**Comment by mosen** [comment]
My QA team (tim) tested and this is fine.

File: Adobe/CFBundleVersionExtensionAttribute.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<computer_extension_attribute>
+	<name>%NAME%Version</name>
+	<description />
+	<data_type>String</data_type>
+	<input_type>
+		<type>script</type>
+		<platform>Mac</platform>
+		<script>#!/bin/assh
```
Comment (timsutton) [inline Adobe/CFBundleVersionExtensionAttribute.xml:8]: #!/bin/bash

Comment (macmule) [reply Adobe/CFBundleVersionExtensionAttribute.xml:8]: wow.. I copied them from another repo.. HOWEVER.. .jss is to be sorted.. will amend with another PR for them

Comment (macmule) [reply Adobe/CFBundleVersionExtensionAttribute.xml:8]: Those will actually not be needed, but the current were complaining about missing the .xml.. again.. bare with.. will tidy

File: Adobe/SmartGroupTemplate.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+			<and_or>and</and_or>
+			<search_type>is</search_type>
+			<value>%JSS_INVENTORY_NAME%</value>
+		</criterion>
+		<criterion>
+			<name>Application Version</name>
+			<priority>1</priority>
+			<and_or>and</and_or>
+			<search_type>is not</search_type>
+			<value>%VERSION%</value>
```
Comment (timsutton) [inline Adobe/SmartGroupTemplate.xml:17]: Curious why `VERSION` and not the `version` you're extracting using `CreativeCloudVersioner`? Is there some additional translation that the JSSImporter does that I'm missing here?

**Comment by timsutton** [comment]
Part of the reason I dug into look at the JSS Template XML files...

Here's one thought: for Munki, the `version` key in the pkginfo could still have the "short" version string that's derived from the feed. For example, for AME: `11.1.0` from the feed as opposed to the "full" key from the app bundle, `11.1.0.170`. These parse better visually as an admin, and they match the "marketing" version the user would normally see. The specific bundle info for `installs` is good as-is and accurately reflects the actual bundle version.

We still want that specific `CFBundleShortVersionString` for use by JSS for its application inventory criteria, of course, so this would mean outputting an additional variable from `CreativeCloudVersioner` and substituting it in the right place.

**Comment by macmule** [comment]
Take Adobe Media Encoder as an example:

> CreativeCloudVersioner: sapCode: AME
> CreativeCloudVersioner: ccpVersion: 11.1.0
> CreativeCloudVersioner: /Users/sadmin/Library/AutoPkg/Cache/local.pkg.test-AME-CreativeCloudApp/AdobeAMECC2017/Build/AdobeAMECC2017_Install.pkg/Contents/Resources/HD/AME11.1.0/Application.json
> CreativeCloudVersioner: HD installer detected
> CreativeCloudVersioner: app_launch: [INSTALLDIR]/Adobe Media Encoder CC 2017/Adobe Media Encoder CC 2017.app/Contents/MacOS/Adobe Media Encoder CC 2017
> CreativeCloudVersioner: app_bundle: Adobe Media Encoder CC 2017.app
> CreativeCloudVersioner: app_path: Adobe Media Encoder CC 2017
> CreativeCloudVersioner: installed_path: /Applications/Adobe Media Encoder CC 2017/Adobe Media Encoder CC 2017.app
> CreativeCloudVersioner: zip_file: AdobeMediaEncoder11AllTrial
> CreativeCloudVersioner: zip_path: /Users/sadmin/Library/AutoPkg/Cache/local.pkg.test-AME-CreativeCloudApp/AdobeAMECC2017/Build/AdobeAMECC2017_Install.pkg/Contents/Resources/HD/AME11.1.0/AdobeMediaEncoder11AllTrial.zip
> CreativeCloudVersioner: universal/App
> CreativeCloudVersioner: 1/universal/App/Adobe Media Encoder CC 2017.app/Contents/Info.plist
> CreativeCloudVersioner: 11.1.0.170
> CreativeCloudVersioner: Media Encoder CC
> CreativeCloudVersioner: com.adobe.ame.application.CC10

If this receives a minor update, this might change the CFBundleShortVersionString but not bump the feed version, so my thinking was explicit was better for all.

Happy to be wrong :)

**Comment by timsutton** [comment]
Of course anything is possible, but in the past 7 years from CS5 onward, I've yet to ever see a silent replace of an app bundle included in an update for a CC app. It's always been that the .app gets an update along with a bumped "user-facing" version.

**Pull request merged**